diff --git a/moveit_planners/pilz_industrial_motion_planner/include/pilz_industrial_motion_planner/planning_context_base.hpp b/moveit_planners/pilz_industrial_motion_planner/include/pilz_industrial_motion_planner/planning_context_base.hpp
index 0abba27..d30d982 100644
--- a/moveit_planners/pilz_industrial_motion_planner/include/pilz_industrial_motion_planner/planning_context_base.hpp
+++ b/moveit_planners/pilz_industrial_motion_planner/include/pilz_industrial_motion_planner/planning_context_base.hpp
@@ -56,9 +56,9 @@ template <typename GeneratorT>
 class PlanningContextBase : public planning_interface::PlanningContext
 {
 public:
-  PlanningContextBase<GeneratorT>(const std::string& name, const std::string& group,
-                                  const moveit::core::RobotModelConstPtr& model,
-                                  const pilz_industrial_motion_planner::LimitsContainer& limits)
+  PlanningContextBase(const std::string& name, const std::string& group,
+                      const moveit::core::RobotModelConstPtr& model,
+                      const pilz_industrial_motion_planner::LimitsContainer& limits)
     : planning_interface::PlanningContext(name, group)
     , terminated_(false)
     , model_(model)
