diff --git a/CMakeLists.txt b/CMakeLists.txt
index 445b836..dca16d8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -36,7 +36,7 @@ set(assimp_LIBRARIES "${assimp_ABS_LIBRARIES}")
 # These need to be in this order to find header files
 find_package(eigen3_cmake_module REQUIRED)
 find_package(Eigen3 REQUIRED)
-
+find_package(Boost REQUIRED random)
 find_package(ament_cmake REQUIRED)
 find_package(assimp REQUIRED)
 find_package(console_bridge REQUIRED)
