diff --git a/moveit_core/CMakeLists.txt b/moveit_core/CMakeLists.txt
index 7a27cda45..1c1809c9a 100644
--- a/moveit_core/CMakeLists.txt
+++ b/moveit_core/CMakeLists.txt
@@ -12,6 +12,7 @@ find_package(common_interfaces REQUIRED)
 find_package(eigen_stl_containers REQUIRED)
 find_package(Eigen3 REQUIRED)
 find_package(eigen3_cmake_module REQUIRED)
+find_package(Boost REQUIRED random)
 find_package(fcl REQUIRED)
 find_package(generate_parameter_library REQUIRED)
 find_package(geometric_shapes REQUIRED)
diff --git a/moveit_kinematics/CMakeLists.txt b/moveit_kinematics/CMakeLists.txt
index 88472a281..1656093bc 100644
--- a/moveit_kinematics/CMakeLists.txt
+++ b/moveit_kinematics/CMakeLists.txt
@@ -7,6 +7,7 @@ moveit_package()

 find_package(ament_cmake REQUIRED)
 find_package(Eigen3 REQUIRED)
+find_package(Boost REQUIRED random)
 find_package(orocos_kdl REQUIRED)
 find_package(tf2_kdl REQUIRED)
 find_package(kdl_parser REQUIRED)
diff --git a/moveit_planners/chomp/chomp_interface/CMakeLists.txt b/moveit_planners/chomp/chomp_interface/CMakeLists.txt
index f356ede55..12e0d042b 100644
--- a/moveit_planners/chomp/chomp_interface/CMakeLists.txt
+++ b/moveit_planners/chomp/chomp_interface/CMakeLists.txt
@@ -11,6 +11,7 @@ find_package(moveit_core REQUIRED)
 find_package(chomp_motion_planner REQUIRED)
 find_package(pluginlib REQUIRED)
 find_package(rclcpp REQUIRED)
+find_package(Boost REQUIRED random)

 set(THIS_PACKAGE_INCLUDE_DEPENDS
   moveit_core
@@ -30,7 +31,7 @@ add_library(${MOVEIT_LIB_NAME} SHARED
   src/chomp_interface.cpp
   src/chomp_planning_context.cpp)
 set_target_properties(${MOVEIT_LIB_NAME} PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
-ament_target_dependencies(${MOVEIT_LIB_NAME} ${THIS_PACKAGE_INCLUDE_DEPENDS})
+ament_target_dependencies(${MOVEIT_LIB_NAME} ${THIS_PACKAGE_INCLUDE_DEPENDS} Boost)

 add_library(moveit_chomp_planner_plugin SHARED src/chomp_plugin.cpp)
 ament_target_dependencies(moveit_chomp_planner_plugin ${THIS_PACKAGE_INCLUDE_DEPENDS})
diff --git a/moveit_planners/chomp/chomp_motion_planner/CMakeLists.txt b/moveit_planners/chomp/chomp_motion_planner/CMakeLists.txt
index 7955acc42..365bc2745 100644
--- a/moveit_planners/chomp/chomp_motion_planner/CMakeLists.txt
+++ b/moveit_planners/chomp/chomp_motion_planner/CMakeLists.txt
@@ -7,6 +7,7 @@ moveit_package()

 # find dependencies
 find_package(ament_cmake REQUIRED)
+find_package(Boost REQUIRED random)
 find_package(moveit_core REQUIRED)
 find_package(rclcpp REQUIRED)
 find_package(trajectory_msgs REQUIRED)
diff --git a/moveit_planners/chomp/chomp_optimizer_adapter/CMakeLists.txt b/moveit_planners/chomp/chomp_optimizer_adapter/CMakeLists.txt
index 113e30b92..2ace5d5b2 100644
--- a/moveit_planners/chomp/chomp_optimizer_adapter/CMakeLists.txt
+++ b/moveit_planners/chomp/chomp_optimizer_adapter/CMakeLists.txt
@@ -3,6 +3,7 @@ project(moveit_chomp_optimizer_adapter)

 # Common cmake code applied to all moveit packages
 find_package(moveit_common REQUIRED)
+find_package(Boost REQUIRED random)
 moveit_package()

 # find dependencies
@@ -20,7 +21,7 @@ set(THIS_PACKAGE_INCLUDE_DEPENDS

 add_library(${PROJECT_NAME} SHARED src/chomp_optimizer_adapter.cpp)
 set_target_properties(${PROJECT_NAME} PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
-ament_target_dependencies(${PROJECT_NAME} ${THIS_PACKAGE_INCLUDE_DEPENDS})
+ament_target_dependencies(${PROJECT_NAME} ${THIS_PACKAGE_INCLUDE_DEPENDS} Boost)

 install(
   TARGETS ${PROJECT_NAME}
diff --git a/moveit_planners/ompl/CMakeLists.txt b/moveit_planners/ompl/CMakeLists.txt
index 4a4ac87f2..4ffe19918 100644
--- a/moveit_planners/ompl/CMakeLists.txt
+++ b/moveit_planners/ompl/CMakeLists.txt
@@ -5,7 +5,7 @@ project(moveit_planners_ompl)
 find_package(moveit_common REQUIRED)
 moveit_package()

-find_package(Boost REQUIRED system filesystem date_time thread serialization)
+find_package(Boost REQUIRED system filesystem date_time thread serialization random)
 find_package(moveit_core REQUIRED)
 find_package(moveit_msgs REQUIRED)
 find_package(moveit_ros_planning REQUIRED)
diff --git a/moveit_planners/pilz_industrial_motion_planner/CMakeLists.txt b/moveit_planners/pilz_industrial_motion_planner/CMakeLists.txt
index e7fcaa351..07a353768 100644
--- a/moveit_planners/pilz_industrial_motion_planner/CMakeLists.txt
+++ b/moveit_planners/pilz_industrial_motion_planner/CMakeLists.txt
@@ -26,7 +26,7 @@ find_package(tf2_kdl REQUIRED)
 find_package(tf2_ros REQUIRED)

 find_package(orocos_kdl REQUIRED)
-find_package(Boost REQUIRED COMPONENTS)
+find_package(Boost REQUIRED random)

 # TODO(henning): Remove/Port
 # if(CATKIN_ENABLE_TESTING AND ENABLE_COVERAGE_TESTING)
diff --git a/moveit_planners/pilz_industrial_motion_planner_testutils/CMakeLists.txt b/moveit_planners/pilz_industrial_motion_planner_testutils/CMakeLists.txt
index 16f1aa6eb..82eb719bd 100644
--- a/moveit_planners/pilz_industrial_motion_planner_testutils/CMakeLists.txt
+++ b/moveit_planners/pilz_industrial_motion_planner_testutils/CMakeLists.txt
@@ -12,7 +12,7 @@ find_package(moveit_msgs REQUIRED)
 find_package(tf2_eigen REQUIRED)
 find_package(eigen3_cmake_module REQUIRED)
 find_package(Eigen3 REQUIRED)
-find_package(Boost REQUIRED COMPONENTS)
+find_package(Boost REQUIRED random)

 ###########
 ## Build ##
diff --git a/moveit_planners/test_configs/prbt_ikfast_manipulator_plugin/CMakeLists.txt b/moveit_planners/test_configs/prbt_ikfast_manipulator_plugin/CMakeLists.txt
index ac35eb601..3114ba391 100644
--- a/moveit_planners/test_configs/prbt_ikfast_manipulator_plugin/CMakeLists.txt
+++ b/moveit_planners/test_configs/prbt_ikfast_manipulator_plugin/CMakeLists.txt
@@ -25,6 +25,7 @@ find_package(tf2_kdl REQUIRED)
 find_package(tf2_eigen REQUIRED)
 find_package(tf2_eigen_kdl REQUIRED)
 find_package(tf2_geometry_msgs REQUIRED)
+find_package(Boost REQUIRED random)

 include_directories(include)

diff --git a/moveit_plugins/moveit_ros_control_interface/CMakeLists.txt b/moveit_plugins/moveit_ros_control_interface/CMakeLists.txt
index dd6d4a79e..53be91d79 100644
--- a/moveit_plugins/moveit_ros_control_interface/CMakeLists.txt
+++ b/moveit_plugins/moveit_ros_control_interface/CMakeLists.txt
@@ -2,6 +2,7 @@ cmake_minimum_required(VERSION 3.22)
 project(moveit_ros_control_interface)

 find_package(ament_cmake REQUIRED)
+find_package(Boost REQUIRED random)

 set(THIS_PACKAGE_INCLUDE_DEPENDS
     rclcpp_action
diff --git a/moveit_plugins/moveit_simple_controller_manager/CMakeLists.txt b/moveit_plugins/moveit_simple_controller_manager/CMakeLists.txt
index 5276dbc90..6855d7569 100644
--- a/moveit_plugins/moveit_simple_controller_manager/CMakeLists.txt
+++ b/moveit_plugins/moveit_simple_controller_manager/CMakeLists.txt
@@ -13,6 +13,7 @@ find_package(control_msgs REQUIRED)
 find_package(rclcpp_action REQUIRED)

 # Finds Boost Components
+find_package(Boost REQUIRED random)
 include(ConfigExtras.cmake)

 set(THIS_PACKAGE_INCLUDE_DEPENDS
diff --git a/moveit_ros/benchmarks/CMakeLists.txt b/moveit_ros/benchmarks/CMakeLists.txt
index 1ed38addc..545841d33 100644
--- a/moveit_ros/benchmarks/CMakeLists.txt
+++ b/moveit_ros/benchmarks/CMakeLists.txt
@@ -16,6 +16,7 @@ find_package(moveit_ros_warehouse REQUIRED)
 find_package(pluginlib REQUIRED)

 # Finds Boost Components
+find_package(Boost REQUIRED random)
 include(ConfigExtras.cmake)

 include_directories(include)
diff --git a/moveit_ros/hybrid_planning/CMakeLists.txt b/moveit_ros/hybrid_planning/CMakeLists.txt
index 504741c51..756e1145f 100644
--- a/moveit_ros/hybrid_planning/CMakeLists.txt
+++ b/moveit_ros/hybrid_planning/CMakeLists.txt
@@ -19,6 +19,7 @@ find_package(std_msgs REQUIRED)
 find_package(std_srvs REQUIRED)
 find_package(tf2_ros REQUIRED)
 find_package(trajectory_msgs REQUIRED)
+find_package(Boost REQUIRED random)

 set(LIBRARIES
   # Components
@@ -46,6 +47,7 @@ set(THIS_PACKAGE_INCLUDE_DEPENDS
   std_srvs
   tf2_ros
   trajectory_msgs
+  Boost
 )

 set(THIS_PACKAGE_INCLUDE_DIRS
diff --git a/moveit_ros/move_group/CMakeLists.txt b/moveit_ros/move_group/CMakeLists.txt
index ebabb9d20..dd0277c92 100644
--- a/moveit_ros/move_group/CMakeLists.txt
+++ b/moveit_ros/move_group/CMakeLists.txt
@@ -17,6 +17,7 @@ find_package(tf2_ros REQUIRED)
 find_package(tf2_geometry_msgs REQUIRED)

 # Finds Boost Components
+find_package(Boost REQUIRED random)
 include(ConfigExtras.cmake)

 set(THIS_PACKAGE_INCLUDE_DEPENDS
@@ -31,6 +32,7 @@ set(THIS_PACKAGE_INCLUDE_DEPENDS
   moveit_ros_occupancy_map_monitor
   moveit_ros_planning
   pluginlib
+  Boost
 )

 include_directories(include)
diff --git a/moveit_ros/moveit_servo/CMakeLists.txt b/moveit_ros/moveit_servo/CMakeLists.txt
index 48621bc8d..d35221589 100644
--- a/moveit_ros/moveit_servo/CMakeLists.txt
+++ b/moveit_ros/moveit_servo/CMakeLists.txt
@@ -29,6 +29,7 @@ set(FAKE_SERVO_CMDS_NAME fake_command_publisher)
 # Common cmake code applied to all moveit packages
 find_package(moveit_common REQUIRED)
 moveit_package()
+find_package(Boost REQUIRED random)

 set(THIS_PACKAGE_INCLUDE_DEPENDS
   control_msgs
@@ -45,6 +46,7 @@ set(THIS_PACKAGE_INCLUDE_DEPENDS
   std_srvs
   tf2_eigen
   trajectory_msgs
+  Boost
 )

 find_package(ament_cmake REQUIRED)
diff --git a/moveit_ros/occupancy_map_monitor/CMakeLists.txt b/moveit_ros/occupancy_map_monitor/CMakeLists.txt
index 3549e1dc5..86c38f6e2 100644
--- a/moveit_ros/occupancy_map_monitor/CMakeLists.txt
+++ b/moveit_ros/occupancy_map_monitor/CMakeLists.txt
@@ -14,6 +14,7 @@ find_package(moveit_core REQUIRED)
 find_package(moveit_msgs REQUIRED)
 find_package(pluginlib REQUIRED)
 find_package(Eigen3 REQUIRED)
+find_package(Boost REQUIRED random)
 find_package(octomap REQUIRED)
 find_package(geometric_shapes REQUIRED)
 find_package(tf2_ros REQUIRED)
diff --git a/moveit_ros/perception/CMakeLists.txt b/moveit_ros/perception/CMakeLists.txt
index a574c0e6f..0a821a0a2 100644
--- a/moveit_ros/perception/CMakeLists.txt
+++ b/moveit_ros/perception/CMakeLists.txt
@@ -52,6 +52,7 @@ find_package(sensor_msgs REQUIRED)
 find_package(moveit_msgs REQUIRED)
 find_package(moveit_ros_occupancy_map_monitor REQUIRED)
 find_package(Eigen3 REQUIRED)
+find_package(Boost REQUIRED random)
 find_package(OpenMP REQUIRED)
 find_package(OpenCV)

@@ -89,6 +90,7 @@ set(THIS_PACKAGE_INCLUDE_DEPENDS
   sensor_msgs
   tf2_geometry_msgs
   Eigen3
+  Boost
 )

 include_directories(${THIS_PACKAGE_INCLUDE_DIRS})
diff --git a/moveit_ros/planning/CMakeLists.txt b/moveit_ros/planning/CMakeLists.txt
index f4e370fbd..eb261267d 100644
--- a/moveit_ros/planning/CMakeLists.txt
+++ b/moveit_ros/planning/CMakeLists.txt
@@ -19,6 +19,7 @@ find_package(tf2_eigen REQUIRED)
 find_package(tf2_msgs REQUIRED)
 find_package(tf2_ros REQUIRED)
 find_package(Eigen3 REQUIRED)
+find_package(Boost REQUIRED random)
 find_package(moveit_core REQUIRED)
 find_package(ament_index_cpp REQUIRED)
 find_package(moveit_ros_occupancy_map_monitor REQUIRED)
diff --git a/moveit_ros/planning_interface/CMakeLists.txt b/moveit_ros/planning_interface/CMakeLists.txt
index 21e1eca10..45c0a4974 100644
--- a/moveit_ros/planning_interface/CMakeLists.txt
+++ b/moveit_ros/planning_interface/CMakeLists.txt
@@ -27,6 +27,7 @@ find_package(Eigen3 REQUIRED)
 # find_package(eigenpy REQUIRED)

 # Finds Boost Components
+find_package(Boost REQUIRED random)
 include(ConfigExtras.cmake)

 set(THIS_PACKAGE_INCLUDE_DIRS
diff --git a/moveit_ros/robot_interaction/CMakeLists.txt b/moveit_ros/robot_interaction/CMakeLists.txt
index 30f89c052..92ab7fd24 100644
--- a/moveit_ros/robot_interaction/CMakeLists.txt
+++ b/moveit_ros/robot_interaction/CMakeLists.txt
@@ -13,7 +13,7 @@ find_package(tf2_eigen REQUIRED)
 find_package(tf2_geometry_msgs REQUIRED)
 find_package(tf2_ros REQUIRED)
 find_package(rclcpp REQUIRED)
-
+find_package(Boost REQUIRED random)

 set(MOVEIT_LIB_NAME moveit_robot_interaction)

@@ -22,6 +22,7 @@ set(THIS_PACKAGE_INCLUDE_DEPENDS
   moveit_ros_planning
   interactive_markers
   tf2_geometry_msgs
+  Boost
 )

 include_directories(include)
diff --git a/moveit_ros/visualization/CMakeLists.txt b/moveit_ros/visualization/CMakeLists.txt
index 0f4a73824..c9eb07a50 100644
--- a/moveit_ros/visualization/CMakeLists.txt
+++ b/moveit_ros/visualization/CMakeLists.txt
@@ -34,6 +34,7 @@ find_package(rviz_default_plugins REQUIRED)
 find_package(rviz_ogre_vendor REQUIRED)

 # Finds Boost Components
+find_package(Boost REQUIRED random)
 include(ConfigExtras.cmake)

 # Qt Stuff
@@ -85,6 +86,7 @@ set(THIS_PACKAGE_INCLUDE_DEPENDS
   rviz_ogre_vendor
   rviz_common
   rviz_default_plugins
+  Boost
 )

 include_directories(rviz_plugin_render_tools/include
diff --git a/moveit_ros/warehouse/CMakeLists.txt b/moveit_ros/warehouse/CMakeLists.txt
index 47f22a9b6..ab5ee1823 100644
--- a/moveit_ros/warehouse/CMakeLists.txt
+++ b/moveit_ros/warehouse/CMakeLists.txt
@@ -12,6 +12,7 @@ find_package(warehouse_ros REQUIRED)
 find_package(moveit_ros_planning REQUIRED)
 find_package(tf2_eigen REQUIRED)
 find_package(tf2_ros REQUIRED)
+find_package(Boost REQUIRED random)

 # Finds Boost Components
 include(ConfigExtras.cmake)
diff --git a/moveit_setup_assistant/moveit_setup_app_plugins/CMakeLists.txt b/moveit_setup_assistant/moveit_setup_app_plugins/CMakeLists.txt
index a5a9fbb07..86ac6a30e 100644
--- a/moveit_setup_assistant/moveit_setup_app_plugins/CMakeLists.txt
+++ b/moveit_setup_assistant/moveit_setup_app_plugins/CMakeLists.txt
@@ -12,6 +12,7 @@ find_package(moveit_ros_visualization REQUIRED)
 find_package(moveit_setup_framework REQUIRED)
 find_package(pluginlib REQUIRED)
 find_package(rclcpp REQUIRED)
+find_package(Boost REQUIRED random)

 # Instruct CMake to run moc automatically when needed.
 set(CMAKE_AUTOMOC ON)
@@ -41,6 +42,7 @@ ament_target_dependencies(${PROJECT_NAME}
   moveit_setup_framework
   pluginlib
   rclcpp
+  Boost
 )

 if(BUILD_TESTING)
diff --git a/moveit_setup_assistant/moveit_setup_assistant/CMakeLists.txt b/moveit_setup_assistant/moveit_setup_assistant/CMakeLists.txt
index 2c3076393..535f945df 100644
--- a/moveit_setup_assistant/moveit_setup_assistant/CMakeLists.txt
+++ b/moveit_setup_assistant/moveit_setup_assistant/CMakeLists.txt
@@ -8,7 +8,7 @@ moveit_package()
 # find dependencies
 find_package(ament_cmake REQUIRED)
 find_package(ament_index_cpp REQUIRED)
-find_package(Boost REQUIRED program_options)
+find_package(Boost REQUIRED program_options random)
 find_package(moveit_setup_framework REQUIRED)
 find_package(moveit_setup_srdf_plugins REQUIRED)
 find_package(pluginlib REQUIRED)
diff --git a/moveit_setup_assistant/moveit_setup_controllers/CMakeLists.txt b/moveit_setup_assistant/moveit_setup_controllers/CMakeLists.txt
index 9c41f06be..0474e963f 100644
--- a/moveit_setup_assistant/moveit_setup_controllers/CMakeLists.txt
+++ b/moveit_setup_assistant/moveit_setup_controllers/CMakeLists.txt
@@ -11,6 +11,7 @@ find_package(ament_index_cpp REQUIRED)
 find_package(moveit_setup_framework REQUIRED)
 find_package(pluginlib REQUIRED)
 find_package(rclcpp REQUIRED)
+find_package(Boost REQUIRED random)

 # Instruct CMake to run moc automatically when needed.
 set(CMAKE_AUTOMOC ON)
@@ -46,6 +47,7 @@ ament_target_dependencies(${PROJECT_NAME}
   moveit_setup_framework
   pluginlib
   rclcpp
+  Boost
 )

 if(BUILD_TESTING)
diff --git a/moveit_setup_assistant/moveit_setup_core_plugins/CMakeLists.txt b/moveit_setup_assistant/moveit_setup_core_plugins/CMakeLists.txt
index 9b54b7b7f..fa08e31c1 100644
--- a/moveit_setup_assistant/moveit_setup_core_plugins/CMakeLists.txt
+++ b/moveit_setup_assistant/moveit_setup_core_plugins/CMakeLists.txt
@@ -14,6 +14,7 @@ find_package(pluginlib REQUIRED)
 find_package(rclcpp REQUIRED)
 find_package(srdfdom REQUIRED)
 find_package(urdf REQUIRED)
+find_package(Boost REQUIRED random)

 # Instruct CMake to run moc automatically when needed.
 set(CMAKE_AUTOMOC ON)
@@ -46,6 +47,7 @@ ament_target_dependencies(${PROJECT_NAME}
   rclcpp
   srdfdom
   urdf
+  Boost
 )

 if(BUILD_TESTING)
diff --git a/moveit_setup_assistant/moveit_setup_framework/CMakeLists.txt b/moveit_setup_assistant/moveit_setup_framework/CMakeLists.txt
index 36e80da02..b80daebb3 100644
--- a/moveit_setup_assistant/moveit_setup_framework/CMakeLists.txt
+++ b/moveit_setup_assistant/moveit_setup_framework/CMakeLists.txt
@@ -19,6 +19,7 @@ find_package(rviz_common REQUIRED)
 find_package(rviz_rendering REQUIRED)
 find_package(srdfdom REQUIRED)
 find_package(urdf REQUIRED)
+find_package(Boost REQUIRED random)

 set(CMAKE_INCLUDE_CURRENT_DIR ON)

@@ -59,6 +60,7 @@ ament_target_dependencies(${PROJECT_NAME}
   rviz_rendering
   srdfdom
   urdf
+  Boost
 )

 if(BUILD_TESTING)
diff --git a/moveit_setup_assistant/moveit_setup_srdf_plugins/CMakeLists.txt b/moveit_setup_assistant/moveit_setup_srdf_plugins/CMakeLists.txt
index df0c7f104..7637d4bae 100644
--- a/moveit_setup_assistant/moveit_setup_srdf_plugins/CMakeLists.txt
+++ b/moveit_setup_assistant/moveit_setup_srdf_plugins/CMakeLists.txt
@@ -9,6 +9,7 @@ find_package(ament_cmake REQUIRED)
 find_package(ament_cmake_ros REQUIRED)
 find_package(moveit_setup_framework REQUIRED)
 find_package(pluginlib REQUIRED)
+find_package(Boost REQUIRED random)

 qt5_wrap_cpp(MOC_FILES
     include/moveit_setup_srdf_plugins/collision_linear_model.hpp
@@ -53,6 +54,7 @@ target_include_directories(${PROJECT_NAME} PUBLIC
 ament_target_dependencies(${PROJECT_NAME}
   moveit_setup_framework
   pluginlib
+  Boost
 )

 if(BUILD_TESTING)
