diff --git a/capabilities/CMakeLists.txt b/capabilities/CMakeLists.txt
index 276a8a3..a1bad72 100644
--- a/capabilities/CMakeLists.txt
+++ b/capabilities/CMakeLists.txt
@@ -4,7 +4,7 @@ project(moveit_task_constructor_capabilities)
 find_package(ament_cmake REQUIRED)
 
 find_package(fmt REQUIRED)
-find_package(Boost REQUIRED)
+find_package(Boost REQUIRED random)
 find_package(moveit_common REQUIRED)
 moveit_package()
 find_package(moveit_core REQUIRED)
diff --git a/core/CMakeLists.txt b/core/CMakeLists.txt
index 15eaba5..fef8c7f 100644
--- a/core/CMakeLists.txt
+++ b/core/CMakeLists.txt
@@ -3,7 +3,7 @@ project(moveit_task_constructor_core)
 
 find_package(ament_cmake REQUIRED)
 
-find_package(Boost REQUIRED)
+find_package(Boost REQUIRED random)
 find_package(fmt REQUIRED)
 find_package(geometry_msgs REQUIRED)
 find_package(moveit_common REQUIRED)
diff --git a/demo/CMakeLists.txt b/demo/CMakeLists.txt
index 5643c58..984904e 100644
--- a/demo/CMakeLists.txt
+++ b/demo/CMakeLists.txt
@@ -17,6 +17,7 @@ find_package(moveit_task_constructor_core REQUIRED)
 find_package(moveit_task_constructor_msgs REQUIRED)
 find_package(rclcpp REQUIRED)
 find_package(tf2_eigen REQUIRED)
+find_package(Boost REQUIRED random)
 
 set(THIS_PACKAGE_INCLUDE_DEPENDS
 	moveit_core
@@ -26,6 +27,7 @@ set(THIS_PACKAGE_INCLUDE_DEPENDS
 	moveit_task_constructor_msgs
 	rclcpp
 	tf2_eigen
+	Boost
 )
 
 add_library(${PROJECT_NAME}_pick_place_task SHARED src/pick_place_task.cpp)
diff --git a/visualization/CMakeLists.txt b/visualization/CMakeLists.txt
index a220b3a..281d786 100644
--- a/visualization/CMakeLists.txt
+++ b/visualization/CMakeLists.txt
@@ -3,7 +3,7 @@ project(moveit_task_constructor_visualization)
 
 find_package(ament_cmake REQUIRED)
 find_package(fmt REQUIRED)
-find_package(Boost REQUIRED)
+find_package(Boost REQUIRED random)
 find_package(moveit_common REQUIRED)
 moveit_package()
 find_package(moveit_core REQUIRED)
